crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -321,41 +336,64 @@ impl PublishFlakeRef {
             Err(ConvertFlakeRefError::LocalFlakeDirty)?;
         }
 
+        // Create a handle to the git repo
         let repo = Git::discover(file_ref.url.path())
             .await
             .map_err(|_| ConvertFlakeRefError::RepoNotFound(file_ref.url.path().into()))?;
 
+        // Get the upstream branch information.
+        // This is essentialy
+        //
+        //   upstream_ref = git rev-parse @{u}
+        //   (remote_name, branch_name) = split_once "/" upstream_ref
+        //   upstream_url = git remote get-url ${remote_name}
+        //   upstream_rev = git ls-remote ${remote_name} ${branch_name}
```

Comment:
This is a helpful comment! I wonder if it belongs on `get_origin()` itself

Reply:
probably, yes